apollo-client/src/main/java/com/ctrip/framework/apollo/internals/DefaultConfig.java

```diff
@@ -61,8 +64,27 @@ public class DefaultConfig extends AbstractConfig implements RepositoryChangeLis
    * @param configRepository the config repository for this config instance
    */
   public DefaultConfig(String namespace, ConfigRepository configRepository) {
+    ConfigUtil configUtil = ApolloInjector.getInstance(ConfigUtil.class);
```

Comment:
This is because the constructor this() must be the first line. I don't want to declare ConfigUtil as static, because non-static will show `Cannot reference 'DefaultConfig.configUtil' before supertype constructor has been called`. Do you have any suggestions?

Reply:
We may pass appid as null and then assign it in the constructor:

```java
  public DefaultConfig(String appId, String namespace, ConfigRepository configRepository) {
    if (appId == null) {
        appId = ApolloInjector.getInstance(ConfigUtil.class).getAppId();
    }

```